### ISSUE TITLE
Strange issue with newly added test files

### ISSUE THREAD

**Issue by ColinEberhardt** [xtuc/webassemblyjs#219]
If I checkout master, and run `git status` it shows changes to a couple of the recently added files:

```
$ git status
On branch decode-name-section
Your branch is up-to-date with 'origin/decode-name-section'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

        modified:   packages/wast-parser/test/tokenizer/number-literals/122_000.11_3_54e0_2_3/actual.wast
        modified:   packages/wast-parser/test/tokenizer/number-literals/122_000.11_3_54e0_2_3/expected.json
```

Looking at the diff for one of these:

```
$ git diff packages/wast-parser/test/tokenizer/number-literals/122_000.11_3_54E0_2_3/actual.wast
diff --git a/packages/wast-parser/test/tokenizer/number-literals/122_000.11_3_54E0_2_3/actual.wast b/packages/wast-parser/test/tokenizer/number-literals/122_000.11_3_54E0_2_3/actual.wast
index 41bd89a..3d5a099 100644
--- a/packages/wast-parser/test/tokenizer/number-literals/122_000.11_3_54E0_2_3/actual.wast
+++ b/packages/wast-parser/test/tokenizer/number-literals/122_000.11_3_54E0_2_3/actual.wast
@@ -1 +1 @@
-122_000.11_3_54E0_2_3
\ No newline at end of file
+122_000.11_3_54e0_2_3
\ No newline at end of file
```

You can see an `E` changed to `e`.

However, when I try to revert (reset hard to HEAD), the changes are still showing.

Any ideas? I've seen line-ending related issues before, but this one has me stumped! 😕

**Comment by maurobringolf** [comment]
These can be so frustrating 😕 IIRC the change from E to e or vice versa happened somewhere during my PR, but other than that I have no clue right now

**Comment by xtuc** [comment]
Yes looks like the recent `toLowerCase` in the tokenizer. I'm not sure why you have theses changes. The CI and my tests are fine tho

**Comment by maurobringolf** [comment]
I checked out `master` and `decode-name-section` without issues, all tests passing locally. Maybe delete the files and then check out again? Not sure

**Comment by ColinEberhardt** [comment]
I'm pretty sure we have a problem here - just tried a 'clean' clone, where `git status` immediately reports those files as having changed:

```
$ git clone https://github.com/xtuc/webassemblyjs.git
Cloning into 'webassemblyjs'...
remote: Counting objects: 13515, done.
remote: Compressing objects: 100% (273/273), done.
remote: Total 13515 (delta 169), reused 349 (delta 132), pack-reused 13092
Receiving objects: 100% (13515/13515), 2.39 MiB | 86.00 KiB/s, done.
Resolving deltas: 100% (6871/6871), done.

$ cd webassemblyjs/

$ git status
On branch master
Your branch is up-to-date with 'origin/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   packages/wast-parser/test/tokenizer/number-literals/122_000.11_3_54E0_2_3/actual.wast
	modified:   packages/wast-parser/test/tokenizer/number-literals/122_000.11_3_54E0_2_3/expected.json

no changes added to commit (use "git add" and/or "git commit -a")
```

**Comment by xtuc** [comment]
Strange, I have done the same and I don't see the changes.

**Comment by ColinEberhardt** [comment]
The issue is the following folders:

![image](https://user-images.githubusercontent.com/1098110/38099440-815a8f6e-3372-11e8-9134-68a1af469216.png)

When cloning the repo on a case-insensitive file system, these collide.

(had to revert to our internal slack channel to solve this one - had me totally stumped!)

**Comment by xtuc** [comment]
Ah, right!

> (had to revert to our internal slack channel to solve this one - had me totally stumped!)

Sorry i'm not sure to understand that?

**Comment by ColinEberhardt** [comment]
> Sorry i'm not sure to understand that?

Ha ha, just meant that I wasn't smart enough to solve this one myself, so had to call on some colleagues ;-)

**Comment by xtuc** [comment]
aha no worries, I personally didn't help I think :)

**Comment by maurobringolf** [comment]
Ah damn. These are annoying, Ive had issues with it too in the past. Any advice on best practices concerning Git and case sensitiveness?

**Comment by ColinEberhardt** [comment]
It's back again!

```
$ git status
On branch master
Your branch is up-to-date with 'origin/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   packages/wast-parser/test/tokenizer/number-literals/generated/122_000-dot-11_3_54e0_2_3/actual.wast
	modified:   packages/wast-parser/test/tokenizer/number-literals/generated/122_000-dot-11_3_54e0_2_3/expected.json

```

**Comment by ColinEberhardt** [comment]
Ah, found it ... this commit https://github.com/xtuc/webassemblyjs/commit/5df3a403d29cbb2525679b325203652efce23617#diff-a30db4834909f5acf9be1c044757bf43 reverts the fix I made in #222 where upper and lowercase versions of the test are placed in subfolders, rather than relying on case sensitive paths

**Comment by ColinEberhardt** [comment]
no idea how we stop this happening again - perhaps there is a CI check that could do a clean checkout then test for unstaged changes?
